### PR TITLE
Update build information in BUILD.md

### PR DESCRIPTION
**Pull request by masterleinad**
Addresses #2263, supersedes #2420.

- `HPX` is an experimental backend
- `ROCm` and `QTHREADS` are removed
- `SEPARATE_LIBS` is removed
- `Kokkos_ENABLE_LARGE_MEM_TESTS` was missing
- `KOKKOS_ARCH_KEPLER` and `Kokkos_ARCH_MAXWELL` are defined through specializations
- `KOKKOS_ARCH_NONE` does not exist
- `HPX` is not a device execution space

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -181,21 +181,15 @@ Options can be enabled by specifiying `-DKokkos_ENABLE_X`.
...
 ## Third-party Libraries (TPLs)
 The following options control activating and locating varial options TPLs.
 
 * Kokkos_CUDA_DIR
     * Location of CUDA library
-    * PATH Default: 
-* Kokkos_ENABLE_HPX
-    * Whether to enable the HPX library
-    * BOOL Default: OFF
+    * PATH Default:
```
Comment (dalg24) [inline BUILD.md]: Default is empty?

**Review by dalg24 — COMMENTED** [review]
> KOKKOS_ARCH_KEPLER and Kokkos_ARCH_MAXWELL are defined through specializations

What do you mean?

**Comment by masterleinad** [comment]
`KOKKOS_ARCH_KEPLER` is defined if `Kokkos_ARCH_KEPLER3?=ON`. This is not a cache variable.

File: BUILD.md (hunk 1 above)
Comment (masterleinad) [reply BUILD.md]: Yes, just as for, e.g. `CUDA`.

**Review by dalg24 — DISMISSED** [review]

File: cmake/KokkosCore_config.h.in [hunk 2]
```diff
@@ -58,7 +58,6 @@
 
 #cmakedefine KOKKOS_COMPILER_CUDA_VERSION @KOKKOS_COMPILER_CUDA_VERSION@
 
-#cmakedefine KOKKOS_ARCH_NONE
```
Comment (dalg24) [inline cmake/KokkosCore_config.h.in]: Please remind me why we had this one.

**Review by dalg24 — COMMENTED** [review]
This PR claims to update the doc in `BUILD.md` but it is really changing 4 files, two of which are also edited in #2470.

File: cmake/KokkosCore_config.h.in (hunk 2 above)
Comment (masterleinad) [reply cmake/KokkosCore_config.h.in]: It became unused after https://github.com/kokkos/kokkos/issues/2241.

Comment (masterleinad) [reply cmake/KokkosCore_config.h.in]: It became unused after #2241.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**